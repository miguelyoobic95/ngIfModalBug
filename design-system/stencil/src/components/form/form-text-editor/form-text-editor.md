---
name: Text Editor
category: Form Components
---

Text-Editor allows you to have a rich text editor directly in a web view

```yoo-form-text-editor.html
     <yoo-form-text-editor></yoo-form-text-editor>
```