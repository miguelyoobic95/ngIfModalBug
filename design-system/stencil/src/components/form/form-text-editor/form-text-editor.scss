

.container {
    width: 100%;
    height: 400px;
    border: 1px solid cssvar(dark-20);
}

:host {
    display: block;
}