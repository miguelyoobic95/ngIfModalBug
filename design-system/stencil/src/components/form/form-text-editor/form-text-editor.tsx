import { Component, Element } from '@stencil/core';
import tinymce from 'tinymce';

@Component({
    tag: 'yoo-form-text-editor',
    styleUrl: 'form-text-editor.scss',
    scoped: true
})
export class YooFormTextEditorComponent {

    protected static componentCounter: number = 0;

    @Element() protected host: HTMLElement;

    private componentId: string;

    constructor() {
        YooFormTextEditorComponent.componentCounter += 1;
        this.componentId = 'mytexteditor_' + YooFormTextEditorComponent.componentCounter;
    }

    componentDidLoad() {
        tinymce.init({
            selector: '#' + this.componentId,
            inline: true,
            theme: 'modern',
            theme_url: 'assets/tinymce/modern/theme.min.js',
            skin: 'light',
            skin_url: 'assets/tinymce/light',
            menubar: false,
            statusbar: false,
            toolbar_items_size: 'small'
        });

    }

    render() {
        return (
            <div class="container" id={this.componentId}></div>
        );
    }
}
