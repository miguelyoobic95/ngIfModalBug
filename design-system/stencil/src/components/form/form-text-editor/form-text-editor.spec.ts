jest.mock('tinymce');
import { TestWindow } from '@stencil/core/testing';
import { YooFormTextEditorComponent } from './form-text-editor';
import tinymce from 'tinymce';

describe('YooFormTextEditorComponent', () => {
    it('should build', () => {
        expect(new YooFormTextEditorComponent()).toBeTruthy();
    });

    describe('Snapshot', () => {
        let element;
        beforeEach( async () => {
          let window = new TestWindow();
          element = await window.load({
            components: [YooFormTextEditorComponent],
            html: '<yoo-form-text-editor></yoo-form-text-editor>'
          });
        });
        it ('Should render', async () => {
            expect(tinymce.init).toHaveBeenCalled();

            expect(element).toMatchSnapshot();
        });
    });
});
