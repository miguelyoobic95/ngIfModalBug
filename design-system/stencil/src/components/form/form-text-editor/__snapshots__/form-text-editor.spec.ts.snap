// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooFormTextEditorComponent Snapshot Should render 1`] = `
<yoo-form-text-editor
  class="hydrated"
  data-yoo-form-text-editor-host=""
>
  
  <div
    class="container"
    data-yoo-form-text-editor=""
    id="mytexteditor_2"
  />
</yoo-form-text-editor>
`;
